src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -850,8 +850,17 @@ private static void DebugVerifyHandlingFrame(UIntPtr handlingFrameSP)
                 else
                 {
                     byte* pFilterFunclet = ehClause._filterAddress;
-                    bool shouldInvokeHandler =
-                        InternalCalls.RhpCallFilterFunclet(exception, pFilterFunclet, frameIter.RegisterSet);
+
+                    bool shouldInvokeHandler = false;
+                    try
+                    {
+                        shouldInvokeHandler =
+                            InternalCalls.RhpCallFilterFunclet(exception, pFilterFunclet, frameIter.RegisterSet);
+                    }
+                    catch when (true)
```

Comment:
This generates `warning CS7095: Filter expression is a constant 'true', consider removing the filter`, why not just
```suggestion
                    catch
```

Reply:
https://github.com/dotnet/runtime/blob/61851156c45b824fc9e307f5fd27ce3a955fd55b/src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs#L11-L12

They're not equivalent. Standalone `catch` is actually `catch (Object)` and we don't want to do that because it might be an object from a different type universe.

Reply:
@Suchiman Hmm, the disabling is already there, so how come it generates the warning?

Reply:
@janvorli sorry, that was my bad, i tested this in isolation and missed that this was both a pattern throughout the file as well as suppressed at the top of the file.